catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/impl/MetacardComparator.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.cache.impl;
+
+import ddf.catalog.data.Metacard;
+
+public class MetacardComparator {
+
+    private MetacardComparator() {
+    }
+
+    public static boolean isSame(Metacard cachedMetacard, Metacard latestMetacard) {
+
+        return (cachedMetacard.getAttribute(Metacard.CHECKSUM).toString().equals(latestMetacard.getAttribute(Metacard.CHECKSUM).toString())
+                         &&
+                cachedMetacard.getAttribute(Metacard.MODIFIED)
+                        .toString()
+                        .equals(latestMetacard.getAttribute(Metacard.MODIFIED)
+                                .toString()) &&
+                cachedMetacard.getAttribute(Metacard.RESOURCE_URI)
```

Comment:
If that's the best we can do, then I am a big fan of @coyotesqrl's answer above. 👍